go/types/map_test.go

```diff
@@ -1155,3 +1155,41 @@ func TestMapRemoveLastWhenNotLoaded(t *testing.T) {
 		assert.True(tm.toMap().Equals(nm))
 	}
 }
+
+func TestMapIterFrom(t *testing.T) {
+	assert := assert.New(t)
+
+	test := func(m Map, rangeStart, rangeEnd, expectedStart, expectedEnd Value) {
```

Comment:
I think this test can be more straightforward given `IterFrom` is dumber than `IterRange`... I'd test:

1. starting before the first element
2. starting on the first element
3. starting in the middle, on an element
4. starting in the middle, not on an element
5. starting on the last element
6. starting past the end

And then instead of the rangeEnd/expectedEnd stuff, just a single test that it terminates if you return `true`.

(You could potentially also test that every value > the previous one when iterating, but that's intrinsic in the map implementation.)